apps/console/src/extensions/configs/models/analytics.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export interface AnalyticsConfig {
+    EventPublisherExtension: {
+        init: () => void;
+        publish: (eventId: string, customProperties?: { [key: string]: string }) => void;
```

Comment:
I think it will be better if we can extract `[key: string]: string` into a type since we are using it in the places as well.

Reply:
This has changed in a later PR
https://github.com/wso2/identity-apps/blob/master/apps/console/src/extensions/configs/models/analytics.ts#L23